modules/apps/27-interchain-accounts/types/codec.go

```diff
@@ -84,7 +84,7 @@ func DeserializeCosmosTx(cdc codec.Codec, data []byte, encoding string) ([]sdk.M
 	switch encoding {
 	case EncodingProtobuf:
 		if err := cdc.Unmarshal(data, &cosmosTx); err != nil {
-			return nil, errorsmod.Wrapf(err, "cannot unmarshal CosmosTx with protobuf")
+			return nil, errorsmod.Wrapf(ErrUnknownDataType, "cannot unmarshal CosmosTx with protobuf")
```

Comment:
I can't see a reason why this was returning the raw unmarshal error? Lmk if I'm missing something here.

Reply:
`ErrUnknownDataType` is somewhat confusing to me here. It failed to unmarshal which could be from malformed data

> I can't see a reason why this was returning the raw unmarshal error?

Probably just based on the pattern of, "if you see an error, wrap it". I'm okay using our own error as a base, but we should be emitting the unmarshal information in the err msg

Reply:
yea, that sounds sensible to me. 

was also slightly confused but went with what was currently used for json unmarhalling, this error seems to be slightly overused :smiley: could add a new one in a follow up